src/interface/inference.py

```diff
@@ -0,0 +1,180 @@
+'''Run a REPL or corpus-level inference
+'''
+# pylint: disable=no-member
+import argparse
+import numpy as np
+import os
+import pandas as pd
+import sys
+
+import logging as log
+from tqdm import tqdm
+
+import torch
+
+from src.tasks.tasks import process_sentence, sentence_to_text_field
+from src.utils import config
+from src.utils.utils import load_model_state, check_arg_name
+from src.preprocess import build_tasks, build_indexers
+from src.models import build_model
+
+from allennlp.data import Vocabulary
+from allennlp.data.dataset import Batch
+from allennlp.data import Instance
+from allennlp.nn.util import move_to_device
+
+log.basicConfig(format='%(asctime)s: %(message)s',
+                datefmt='%m/%d %I:%M:%S %p', level=log.INFO)
+
+
+def handle_arguments(cl_arguments):
+    parser = argparse.ArgumentParser(description='')
+    # Configuration files
+    parser.add_argument('--config_file', '-c', type=str, nargs="+",
+                        help="Config file(s) (.conf) for model parameters.")
+    parser.add_argument('--overrides', '-o', type=str, default=None,
+                        help="Parameter overrides, as valid HOCON string.")
+    parser.add_argument('--inference_mode', type=str, default="repl",
+                        help="Running as REPL or process corpus."
+                             " [repl, corpus]")
+    parser.add_argument('--input_path', type=str, default=None,
+                        help="Input path for running inference."
+                             " One input per line."
+                             " Only in eval_mode='corpus'")
+    parser.add_argument('--output_path', type=str, default=None,
+                        help="Output path for running inference."
+                             " Only in eval_mode='corpus'")
+
+    return parser.parse_args(cl_arguments)
+
+
+def main(cl_arguments):
+    ''' Run REPL for a CoLA model '''
+    cl_args = handle_arguments(cl_arguments)
+    args = config.params_from_file(cl_args.config_file, cl_args.overrides)
+
+    # Raise error if obsolete arg names are present
+    check_arg_name(args)
+
+    if args.cuda >= 0:
+        try:
+            if not torch.cuda.is_available():
+                raise EnvironmentError("CUDA is not available, or not detected"
+                                       " by PyTorch.")
+            log.info("Using GPU %d", args.cuda)
+            torch.cuda.set_device(args.cuda)
+        except Exception:
+            log.warning(
+                "GPU access failed. You might be using a CPU-only"
+                " installation of PyTorch. Falling back to CPU."
+            )
+            args.cuda = -1
+
+    # Prepare data #
+    pretrain_tasks, target_tasks, vocab, word_embs = build_tasks(args)
```

Comment:
Skip as much as possible of the preprocessing for any non-CoLA tasks, and consider moving the CoLA assert earlier.